### PR TITLE
Fix tab label disambiguation regression

### PR DESCRIPTION
**Pull request by jeremypw**
At some point tab labels stopped being disambiguated.  This PR moves the diambiguation code to `FileUtils` and further unit tests added to valid the function for some corner cases.

The revised disambiguation code allows losing the `id` property from the ViewContainer class.

### CODE REVIEW

**Comment by danirabbit** [comment]
@jeremypw can you resolve conflicts here?

**Comment by jeremypw** [comment]
@danirabbit Conflict fixed.

**Comment by zeebok** [comment]
@jeremypw Conflicts with main again

File: libcore/tests/UtilTests/UtilTests.vala [hunk 1]
```diff
@@ -346,6 +346,64 @@ void add_file_utils_tests () {
...
+        );
+        assert (result == "B/folder");
+    });
+
+    Test.add_func ("/FileUtils/disambiguate_path/different_length", () => {
+        var result = Files.FileUtils.disambiguate_uri (
+            "/A/B/folder",
+            "X/A/B/folder"
+        );
+        assert (result == "A/B/folder");
```
Comment (zeebok) [inline libcore/tests/UtilTests/UtilTests.vala:363]: Should this actually be "/A/B/folder" for the expected result since it implies an absolute path rather than relative?

**Review by zeebok — APPROVED** [review]
Only have one question otherwise this looks good

File: libcore/tests/UtilTests/UtilTests.vala (hunk 1 above)
Comment (jeremypw) [reply libcore/tests/UtilTests/UtilTests.vala:363]: Yes - you are right!  This function is supposed to return enough  to distinguish the two paths.  I'll fix.

**Review by zeebok — APPROVED** [review]
